### ISSUE TITLE
Error in labextension install

### ISSUE THREAD

**Issue by robmarkcole** [SimonBiggs/scriptedforms#232]
See traceback, could be due to our firewall..?:
```
(C:\ProgramData\Anaconda3) C:\Users\RCole\Downloads\scriptedforms-master\scripte
dforms-master>jupyter labextension install scriptedforms
> node c:\programdata\anaconda3\lib\site-packages\jupyterlab\staging\yarn.js ins
tall
yarn install v1.3.2
[1/4] Resolving packages...
[2/4] Fetching packages...
error An unexpected error occurred: "https://registry.yarnpkg.com/tslib/-/tslib-
1.8.1.tgz: self signed certificate in certificate chain".
info If you think this is a bug, please open a bug report with the information p
rovided in "C:\\Users\\RCole\\Downloads\\scriptedforms-master\\scriptedforms-mas
ter\\scriptedforms\\yarn-error.log".
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this
command.
> "C:\Program Files\nodejs\npm.CMD" pack C:\Users\RCole\Downloads\scriptedforms-
master\scriptedforms-master\scriptedforms
simonbiggs-scriptedforms-0.9.1-dev2.tgz

Errored, use --debug for full output:
ValueError: "C:\Users\RCole\Downloads\scriptedforms-master\scriptedforms-master\
scriptedforms" is not a valid extension:
Missing extension module "src/main.ts.js"
```

**Comment by SimonBiggs** [comment]
No that's expected. That will be the first error you see if you're using the unpatched version of JupyterLab. ScriptedForms doesn't yet work with the released version of JupyterLab. This pull request needs to go through for that to work:

https://github.com/jupyterlab/jupyterlab/pull/4334

To get it to work while that pull request is in transit you will need to download my fork of jupyterlab, swap to the "allow-angular-templates" branch, and then follow the jupyterlab contributor documentation on how to install jupyterlab from source.

**Comment by SimonBiggs** [comment]
If you're interested in the discussion head on over to https://gitter.im/jupyterlab/jupyterlab

**Comment by SimonBiggs** [comment]
Oh and yes, I skipped to the last line of the error log. That last line is due to needing the pull request. But the middle one in there is due to your https chain being broken by an "employer-in-the-middle". https://en.m.wikipedia.org/wiki/Man-in-the-middle_attack

To fix that can you create a config file that declares to the JupyterLab package manager that having the man in the middle is okay. The following comment mentions what to put in the file and where to put it.

https://github.com/yarnpkg/yarn/issues/980#issuecomment-257413314

**Comment by OxygenLithium** [comment]
Do you know if its possible to offline install the allow angular version?
I tried but its missing several dependencies. I found the dependencies online but not sure where JupyterLab is looking for them locally (if at all)

Any suggestions?

**Comment by OxygenLithium** [comment]
Figured it out but would not recommend... requires so many dependencies

**Comment by SimonBiggs** [comment]
Maybe play easy mode. Install it on another computer... then chuck it on a
USB :)

**Comment by OxygenLithium** [comment]
That's the new plan, just waiting on a python install :)

**Comment by SimonBiggs** [comment]
I might install a dev version of jupyterlab into the portable bundle. It
looks like my pull request won't make the release happening in the next
couple of days. So it will have to wait until beta 3 before a standard
python install will work :(
